### PR TITLE
fix: alert flip flops

### PR DESCRIPTION
**Pull request by rramalinga**
## Description

Adding a new field for `successfulTimeStampMap` for every account

## Motivation and Context

Vargid scheduler schedules a bunch of policies for evaluation to NATS server which gets picked by Yeul Worker. The Yeul Worker picks up a job from the NATS server and executes the query for evaluation. It returns the result to Vargid API to aggregate and raise the alerts.

When Vargid API receives the `cloudschedulednetworkquery` response back, it updates the object's `successfulExecutionTimeStamp` with the time that it receives the response. This information is crucial in identifying alerts which are not raised in this successful run (i.e. while alert aggregation, the current set of alerts in this policy run is updated with the latest time stamp and the old alerts are left with the timestamp of the policy run in which it showed up. The resolver uses this information to resolve old alerts)

**Root cause**
However, in a recent optimization, we broke down big policy jobs (with multiple accounts) to smaller policy jobs (with smaller accounts) so that multiple jobs are scheduled for a single big policy job. If all the smaller policy jobs runs successfully, then there should be no problem. 

However, assume a case where you have a policy job **J** broken down into two jobs - **J1** and **J2**. When J1 executes successfully it updates the `successfulExecutionTimeStamp` with the latest timestamp and if J2 fails for some reason (yeul worker fails because of big accounts in J2), then it fails silently. This means that previously-raised alert records corresponding to J2 would be resolved even though there was no successful execution for the policy.

**The fix**
The core issue is we are using one timestamp `successfulExecutionTimeStamp` for the whole job when the accounts are being evaluated separately. We need a data structure to track the last successful execution for every account so that we can resolve alerts based on the `successfulExecutionTimeStamp` for the account associated to the alert.

A new data structure `successfulExecutionTimeStampMap` mapping `accounts to successfulExecTimestamp` is added to the query object and every time a CNQ response is recevied in Vargid API, then we would look at the accounts associated with the job and update the last successful execution time. While resolving alerts, we would look at the `successfulExecTimestamp` corresponding to the account.
## Types of changes

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: specs/cloudschedulednetworkquery.spec [hunk 1]
```diff
@@ -80,6 +80,13 @@ attributes:
     stored: true
     orderable: true
 
+  - name: successfulExecutionTimestampMap
+    description: Mapping of last successful execution timestamp for every account.
+    type: external
+    exposed: true
+    subtype: map[string]string
+    stored: true
```
Comment (t00f) [inline specs/cloudschedulednetworkquery.spec:88]: Can you provide a more detailed description in this PR about the use of the map[string]string? I know we are trying to solve the alert flipping issue, but it looks weird to store that in db

Comment (rramalinga) [reply specs/cloudschedulednetworkquery.spec:88]: Added a note.

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "flx-alert-flips",
    "component": "backend",
    "pr-id": "1801",
    "commit-sha": "3ecb1aa40f32585f5e82ceae42a7f31c14dfe747",
    "pipeline": "master"
  },
  {
    "project": "flx-alert-flips",
    "component": "gaia",
    "pr-id": "820",
    "commit-sha": "de2550a702dd699579f1cda405d2d1e22d00ce5f",
    "pipeline": "master"
  }
]
```

**Review by t00f — APPROVED** [review]

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "flx-alert-flips",
    "component": "backend",
    "pr-id": "1801",
    "commit-sha": "83b4394161f1ce23ccc99cc3f396e51031cd442d",
    "pipeline": "master"
  },
  {
    "project": "flx-alert-flips",
    "component": "gaia",
    "pr-id": "820",
    "commit-sha": "de2550a702dd699579f1cda405d2d1e22d00ce5f",
    "pipeline": "master"
  }
]
```

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "flx-alert-flips",
    "component": "backend",
    "pr-id": "1801",
    "commit-sha": "83b4394161f1ce23ccc99cc3f396e51031cd442d",
    "pipeline": "master"
  },
  {
    "project": "flx-alert-flips",
    "component": "gaia",
    "pr-id": "820",
    "commit-sha": "de2550a702dd699579f1cda405d2d1e22d00ce5f",
    "pipeline": "master"
  }
]
```

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "flx-alert-flips",
    "component": "backend",
    "pr-id": "1801",
    "commit-sha": "f07b2cbc4b99d9f9e1929e05acf9bb6cfcb4ff98",
    "pipeline": "master"
  },
  {
    "project": "flx-alert-flips",
    "component": "gaia",
    "pr-id": "820",
    "commit-sha": "de2550a702dd699579f1cda405d2d1e22d00ce5f",
    "pipeline": "master"
  }
]
```

**Comment by abhijitherekar** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "flx-alert-flips",
    "component": "backend",
    "pr-id": "1801",
    "commit-sha": "f07b2cbc4b99d9f9e1929e05acf9bb6cfcb4ff98",
    "pipeline": "master"
  },
  {
    "project": "flx-alert-flips",
    "component": "gaia",
    "pr-id": "820",
    "commit-sha": "de2550a702dd699579f1cda405d2d1e22d00ce5f",
    "pipeline": "master"
  }
]
```

**Review by Rohit4793 — APPROVED** [review]
Chris already approved it but after was gone after merging master so approving it to unblock Abhi as requested.

**Pull request merged**